filebeat/input/filestream/internal/input-logfile/harvester_test.go

```diff
@@ -47,19 +47,19 @@ func TestReaderGroup(t *testing.T) {
 	}
 
 	t.Run("assert new group is empty", func(t *testing.T) {
-		rg := newReaderGroup()
+		rg := newReaderGroup(0)
 		require.Equal(t, 0, len(rg.table))
 	})
 
 	t.Run("assert non existent key can be removed", func(t *testing.T) {
-		rg := newReaderGroup()
+		rg := newReaderGroup(0)
```

Comment:
just a readability nit, what would you say about having two constructor functions one parameterless anda another one with `WithLimit` suffix. the first one would pass `0` to second one so you don't have to do it all over the place and wonder what this 0 stands for